File: sol/solution/0001-0100/0013. Roman to Integer/Solution.c
int romanToInt(char *s)
{
    int map[128] = {0};
    map['I'] = 1;
    map['V'] = 5;
    map['X'] = 10;
    map['L'] = 50;
    map['C'] = 100;
    map['D'] = 500;
    map['M'] = 1000;
    int res = 0;
    for (int i = 0; s[i]; i++)
    {
        if (s[i + 1] && map[s[i]] < map[s[i + 1]])
            res -= map[s[i]];
        else
            res += map[s[i]];
    }
    return res;
}

File: sol/solution/0001-0100/0013. Roman to Integer/Solution.swift
class Solution {
    func romanToInt(_ s: String) -> Int {
        let val: [Character: Int] = ["I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000]
        let chars = Array(s)
        var res = 0
        for i in 0..<chars.count {
            if i + 1 < chars.count && val[chars[i]]! < val[chars[i+1]]! {
                res -= val[chars[i]]!
            } else {
                res += val[chars[i]]!
            }
        }
        return res
    }
}